syntheseus/cli/eval_single_step.py

```diff
@@ -157,22 +159,22 @@ def get_results(
         time_model_call_end = time.time()
         timing_results["time_model_call"] = time_model_call_end - time_model_call_start
 
-    batch_outputs: List[Sequence[Prediction[InputType, OutputType]]] = []
+    batch_outputs: List[Sequence[ReactionType]] = []
 
     for outputs in raw_batch_outputs:
         if len(outputs) > num_results:
             raise ValueError(f"Requested {num_results} results, but model produced {len(outputs)}")
 
-        seen_outputs: Set[OutputType] = set()
-        selected_predictions: List[Prediction[InputType, OutputType]] = []
+        seen_outputs: Set[ReactionType] = set()
+        selected_predictions: List[ReactionType] = []
 
-        for prediction in outputs:
+        for reaction in outputs:
             if skip_repeats:
-                if prediction.output in seen_outputs:
+                if reaction in seen_outputs:
                     continue
-                seen_outputs.add(prediction.output)
+                seen_outputs.add(reaction)
 
-            selected_predictions.append(prediction)
+            selected_predictions.append(reaction)
```

Comment:
Good attention to detail! I was aware of this when I made the change but don't think it should be an issue.

The `identifier` flag exists in case we want to create two reactions with the same reactants and products which are *not* equal. For example, there may be two ways to perform a reaction which have different costs or yields. Although the costs and yields themselves would be stored in the metadata, the `identifier` flag could be used to distinguish them. 

So far edge cases like this have never come up and the identifier has never been used. Therefore, at least at the moment comparing reactions is just equivalent to comparing reactants and products. When making this change, I considered what should happen if in the future reaction models *do* return reactions with identifiers? These were the key points that led me to include identifiers in the comparison?

1. The principle of an identifier is that two reactions with different identifiers are *not* equal, and there is no reason why the eval script should follow a different set of rules.
2. The ground-truth reactions are inputted as dataset objects which could include identifiers if desired. Perhaps then the proper thing to do in this case is to also supply a better dataset?
3. Identifiers are probably only useful in cases where reaction conditions are important. In that case would anybody even care about the accuracy metrics from this eval script?

What do you think @kmaziarz? I could refactor this (and similar comment below) to just compare reactants/products. I could also add a check for identifiers that are not `None` and raise a `NotImplementedError("Behaviour for reactions with identifiers is not defined")`.

Reply:
I see, makes sense. I think raising an error as you mentioned wouldn't hurt, just to make sure we don't accidentally do something we don't want to?

Reply:
I changed this to check for identifiers and raise an error. Does it look good?